java/concept/ConceptManagerImpl.java

```diff
@@ -83,78 +85,127 @@ public AttributeType getRootAttributeType() {
     @Nullable
     public EntityType getEntityType(String label) {
         if (label == null || label.isEmpty()) throw new TypeDBClientException(MISSING_LABEL);
-        com.vaticle.typedb.client.jni.Concept res = concepts_get_entity_type(nativeTransaction, label);
-        if (res != null) return new EntityTypeImpl(res);
-        else return null;
+        if (!nativeTransaction.isOwned()) throw new TypeDBClientException(TRANSACTION_CLOSED);
+        try {
+            com.vaticle.typedb.client.jni.Concept res = concepts_get_entity_type(nativeTransaction, label);
+            if (res != null) return new EntityTypeImpl(res);
+            else return null;
+        } catch (com.vaticle.typedb.client.jni.Error e) {
```

Comment:
Can we have our `TypeDBClientException` extend the JNI error so we don't have to catch and re-throw all the time?

Reply:
That would require the user to catch `com.vaticle.typedb.client.jni.Error` instead of `com.vaticle.typedb.client.common.exception.TypeDBClientException`, which leaks implementation details.

Unless you meant the other way around (`jni.Error extends TypeDBClientException`), which would be friendlier, but not possible at the moment as that would involve a circular dependency. We could do that at some point by extracting the client error into its own top-level common package (or into typedb-common).

Reply:
Let's keep this for now.